CMakeLists.txt

```diff
@@ -7,6 +7,10 @@ project(box2d VERSION 2.4.0)
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+
+option(BUILD_SHARED_LIBS "Build Box2D as a shared library" OFF)
```

Comment:
You don't need that. This is a global cmake flag.

Reply:
The [documentation](https://cmake.org/cmake/help/v3.8/variable/BUILD_SHARED_LIBS.html) says that it's often an option.

Reply:
Right. But that doesn't mean it's needed. It will work in the same way, with or without this line, that's all I'm saying :)

Reply:
Having an option allows to set a default value (`OFF`) which is the default now.